apps/s_client.c

```diff
@@ -2257,6 +2259,44 @@ int s_client_main(int argc, char **argv)
             }
         }
         break;
+    case PROTO_MYSQL:
+        {
+            /*
+             * https://dev.mysql.com/doc/dev/mysql-server/latest/page_protocol_basic_packets.html
+             * https://dev.mysql.com/doc/internals/en/ssl.html
+             * https://dev.mysql.com/doc/internals/en/connection-phase-packets.html#packet-Protocol::HandshakeResponse
+             */
+            static const unsigned char mysql_ssl_request[] = {
+                /* payload_length,   sequence_id */
+                   0x20, 0x00, 0x00, 0x01,
+                /* payload */
+                /* capability flags, CLIENT_SSL always set */
+                   0x85, 0xae, 0x7f, 0x00,
+                /* max-packet size */
+                   0x00, 0x00, 0x00, 0x01,
+                /* character set */
+                   0x21,
+                /* string[23] reserved (all [0]) */
+                   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+            };
+            int bytes;
+
+            /*
+             * https://dev.mysql.com/doc/dev/mysql-server/latest/page_protocol_connection_phase.html
+             * Receiving Initial Handshake packet from mysql server
+             */
+            bytes = BIO_read(sbio, sbuf, BUFSIZZ);
+            if (bytes < 1)
+                goto shut;
+
+            /* Sending Handshake packet containing CLIENT_SSL to mysql server, to request SSL connection. */
```

Comment:
It looks like server does tell if it supports SSL/TLS. And as already said it would appropriate to parse the just read packet and not enter SSL/TLS negotiation in case server doesn't support it.